lib/Parse/ParseExpr.cpp

```diff
@@ -2621,8 +2622,19 @@ ParserStatus Parser::parseClosureSignatureIfPresent(
         // It is a common error to try to capture a nested field instead of just
         // a local name, reject it with a specific error message.
         if (Tok.isAny(tok::period, tok::exclaim_postfix,tok::question_postfix)){
-          diagnose(Tok, diag::cannot_capture_fields);
-          skipUntil(tok::comma, tok::r_square);
+          auto diag = diagnose(Tok, diag::cannot_capture_fields);
+          while (peekToken().isNot(tok::comma, tok::r_square, tok::eof,
+                                   tok::pound_endif, tok::pound_else,
+                                   tok::pound_elseif))
```

Comment:
Could you add `tok::kw_in` and `tok::r_brace` here? Consider `{ [foo.bar in `  or `{ [foo.bar } ` we don't want to skip over them.

Reply:
`parseClosureSignatureIfPresent` function has a pretty early check to ensure that the square braces are closed.
https://github.com/apple/swift/blob/aaee1c4bcfb7af21c2563b5caa564727a34db3f6/lib/Parse/ParseExpr.cpp#L2477-L2482

I've added the check, but I'm not sure how to actually hit them. (Actually adding them to the closures.swift test causes other warnings to not appear, causing more noise in the test, which is why I didn't add them).

Reply:
Ah true. I didn't remember `parseClosureSignatureIfPresent`, but adding the check doesn't hurt. Thank you!